### PR TITLE
Add ArrayHelper::pathExists

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/ArrayHelper.php [hunk 1]
```diff
@@ -879,26 +879,98 @@ public static function map(array $array, $from, $to, $group = null): array
...
+        }
+
+        $keys = [];
+        foreach (array_keys($array) as $k) {
+            if (strcasecmp($key, (string)$k) === 0) {
+                $keys[] = $k;
+            }
+        }
+
+        return $keys;
```
Comment (xepozz) [inline src/ArrayHelper.php:954]:
```suggestion
        return array_filter(array_keys($array), fn($k) => strcasecmp($key, (string)$k) === 0);
```

File: tests/ArrayHelper/KeyAndPathExistsTest.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            '1.0.status' => 'dev',
+            2 => 'two',
+        ],
+        '42.7' => 500,
+    ];
+
+    /**
+     * @return array[] common test data for [[testKeyExists()]] and [[testPathExists()]]
+     */
+    private function commonData(): array
```
Comment (xepozz) [inline tests/ArrayHelper/KeyAndPathExistsTest.php:43]: Let's remove this method and copy-paste these cases to needy methods

File: src/ArrayHelper.php (hunk 1 above)
Comment (vjik) [reply src/ArrayHelper.php]: Done.

File: tests/ArrayHelper/KeyAndPathExistsTest.php (hunk 2 above)
Comment (vjik) [reply tests/ArrayHelper/KeyAndPathExistsTest.php:43]: This methods should be work different only for strings with dots. In all other cases this methods equally, so I create separately method with common data.

Also made for tests of getValue and getValueByPath methods.

**Review by xepozz — APPROVED** [review]

**Pull request merged**